src/object/cli_ec.c

```diff
@@ -2822,20 +2825,23 @@ obj_recx_ec2_daos(struct daos_oclass_attr *oca, int shard,
 			daos_size_t daos_size;
 			daos_off_t  daos_off;
 
-			daos_off = obj_ec_idx_vos2daos(offset, stripe_nr,
-						       cell_nr, tgt_idx);
-			daos_size = min(roundup(offset + 1, cell_nr) - offset,
-					size);
-			D_ASSERT(idx < total);
+			daos_off = obj_ec_idx_vos2daos(offset, stripe_nr, cell_nr, tgt_idx);
+			daos_size = min(roundup(offset + 1, cell_nr) - offset, size);
+			D_ASSERTF(idx < total, "idx %d total %d recxs "DF_U64"/"DF_U64"\n",
+				  idx, total, recxs[i].rx_idx, recxs[i].rx_nr);
 			tgt_recxs[idx].rx_idx = daos_off;
 			tgt_recxs[idx].rx_nr = daos_size;
 			offset += daos_size;
-			size -= daos_size;
+			if (size > daos_size)
```

Comment:
since "daos_size = min(roundup(offset + 1, cell_nr) - offset, size);", seems can directly 
D_ASSERT(size >= daos_size), 
seems the worried case of negative size is impossible?

Reply:
yes, it seems impossible to make size to be "negative", more about the style fixing for unsigned size. maybe the commit message needs to be updated.